fedmsg/crypto/x509.py

```diff
@@ -113,7 +113,7 @@ def fail(reason):
     for field in ['signature', 'certificate']:
         if field not in message:
             return fail("No %r field found." % field)
-        if not isinstance(message[field], str):
+        if not isinstance(message[field], basestring):
```

Comment:
That's legit.

The underlying issue is (I think) that some routines pass unicode signatures in here, while others pass byte-strings.

I guess I'm with you.. that we should enforce "unicode" as the api with ``six.text_type`` and then go looking for any places that may erroneously call this with byte strings.